File: mdoc/src/main/scala/mdoc/internal/worksheets/EvaluatedWorksheet.scala
package mdoc.internal.worksheets

import java.{util => ju}
import mdoc.{interfaces => i}
import java.nio.file.Path

case class EvaluatedWorksheet(
    val diagnostics: ju.List[i.Diagnostic],
    val statements: ju.List[i.EvaluatedWorksheetStatement],
    val scalacOptions: ju.List[String],
    val classpath: ju.List[Path],
    val dependencies: ju.List[coursierapi.Dependency],
    val repositories: ju.List[coursierapi.Repository]
) extends mdoc.interfaces.EvaluatedWorksheet

File: mdoc-interfaces/src/main/scala/mdoc/interfaces/EvaluatedWorksheet.java
package mdoc.interfaces;

import java.util.List;
import java.nio.file.Path;
import coursierapi.Dependency;
import coursierapi.Repository;

public abstract class EvaluatedWorksheet {

  public abstract List<Diagnostic> diagnostics();
  public abstract List<EvaluatedWorksheetStatement> statements();

  public abstract List<String> scalacOptions();
  public abstract List<Path> classpath();
  public abstract List<Dependency> dependencies();
  public abstract List<Repository> repositories();

}
